### PR TITLE
optional auto disconnect and disconnectAll action

### PR DESCRIPTION
**Pull request by stevejpurves**

### CODE REVIEW

File: src/components/AnchorBase.tsx [hunk 1]
```diff
@@ -35,13 +45,6 @@ export const AnchorBase = (props: Props) => {
       dispatch(connectAnchorBase(parentDoc, anchor, el));
     }
   }, []);
-  useEffect(() => {
-    if (!ref) return () => null;
-    return () => {
-      const parentDoc = getDoc(ref);
-      return dispatch(disconnectAnchor(parentDoc, ref));
-    };
-  }, [ref]);
```
Comment (stevejpurves) [inline src/components/AnchorBase.tsx:44]: There were two independent UseEffects to trigger the disconnect here, which is double trouble in terms of the number of action issues on unmount. @rowanc1 does the change to a single still capture the cases covered by the original pair?

Comment (rowanc1) [reply src/components/AnchorBase.tsx:44]: Yeah - I think you adding the above seems to work. Did you try it?

File: src/components/AnchorBase.tsx [hunk 2]
```diff
@@ -7,24 +7,34 @@ import { isSidenoteSelected } from '../store/ui/selectors';
...
   const [doc, setDoc] = useState<string>();
   const [ref, setRef] = useState<HTMLDivElement | null>(null);
 
   useEffect(() => {
-    if (ref == null || doc == null) return () => {};
-    return () => dispatch(disconnectAnchor(doc, ref));
-  }, [doc, ref]);
+    if (!ref || !autoDisconnect) return () => null;
+    return () => {
+      const parentDoc = doc ?? getDoc(ref);
```
Comment (rowanc1) [inline src/components/AnchorBase.tsx:34]: This seems to cover the use case you removed below. In one action. So should be good.

File: src/components/AnchorBase.tsx (hunk 1 above)
Comment (stevejpurves) [reply src/components/AnchorBase.tsx:44]: yes, various times :p

File: src/store/ui/reducers.ts [hunk 3]
```diff
@@ -106,6 +107,12 @@ const uiReducer = (
         },
       };
     }
+    case UI_DISCONNECT_ALL_ANCHORS: {
```
Comment (rowanc1) [inline src/store/ui/reducers.ts:110]: Is this reducer actually more like reset sidenote state?

Just looking at it it does more than the anchors, also removes all comments across all docs?

Comment (stevejpurves) [reply src/store/ui/reducers.ts:110]: yes is takes out the full state

File: src/components/AnchorBase.tsx [hunk 4]
```diff
@@ -52,6 +43,7 @@ export const AnchorBase = (props: Props) => {
 
 AnchorBase.defaultProps = {
   className: undefined,
+  refs: undefined,
```
Comment (rowanc1) [inline src/components/AnchorBase.tsx:46]: Not sure what this is?

File: src/store/ui/actions.ts [hunk 5]
```diff
@@ -104,6 +106,15 @@ export function disconnectAnchor(
   };
 }
 
+export function resetAll(): AppThunk<void> {
+  return (dispatch) => {
+    dispatch({
+      type: UI_RESET_ALL,
```
Comment (rowanc1) [inline src/store/ui/actions.ts:112]: We probably want a `SIDENOTES` in there. These actions could conflict with other apps.

File: src/store/ui/types.ts [hunk 6]
```diff
@@ -55,6 +56,10 @@ export interface DisconnectAnchorAction {
   };
 }
 
+export interface DisconnectAllAnchors {
```
Comment (rowanc1) [inline src/store/ui/types.ts:59]: Rename!

**Review by rowanc1 — COMMENTED** [review]
Minor comments. Sorry - still asking to rename things.

**Review by rowanc1 — APPROVED** [review]
Looks good to me!

**Pull request merged**